### ISSUE TITLE
Should DLSS 3 Frame Generation work today?

### ISSUE THREAD

**Issue by philipl** [jp7677/dxvk-nvapi#103]
I've been trying to work out if the existing DLSS support is supposed to be enough to enable frame generation? There hasn't been any unambiguous statement from nvidia. I've been trying out Portal RTX on a 4090 and the Frame Generation option is not exposed, although the rest of the DLSS functionality works just fine.

Is there additional work nvidia needs to do? Is this related to the fact that dxvk-nvapi reports Ampere instead of Ada as noted in the 0.6 release notes - although that says its only for DLSS 2.0 -> 2.4 and Portal RTX should be using 3.0? or something else?

Thanks!

**Comment by jp7677** [comment]
I don’t know how exactly how DLSS3 works (tbf, I don’t know the details about DLSS2 either :)), but I’m pretty certain that getting support for it would still need work on several places in the stack. So yes, at least from my limited knowledge, DLSS3 not being available should be the current state.

For reference, bringing DLSS2 to Linux/Proton resulted in work in the NVIDIA driver, in DXVK/VKD3D-Proton, in this project. But I don’t know if the same applies to DLSS3. From my understanding DLSS3 is also a very different kind of beast than DLSS2.

Yeah, the spoofing work around is really just for a certain range of DLSS2 version, newer versions know about Ada and for those Ada architecture will be reported.

**Comment by adamnv** [comment]
> I’m pretty certain that getting support for it would still need work on several places in the stack. So yes, at least from my limited knowledge, DLSS3 not being available should be the current state.

Correct! :)

**Comment by oscarbg** [comment]
@jp7677 Curious if new NV 530 Linux driver branch has some progress toward supporting DLSS3 frame generarion on Proton?

**Comment by philipl** [comment]
@adamnv is nvidia working on exposing Frame Gen on Linux?

**Comment by eXt73** [comment]
> I wouldn't be able to say...

So You don't know anything about it ? Nvidia is 'playing' with us ;/

**Comment by jp7677** [comment]
@eXt73 Please refrain from making nonsensical comments.

**Comment by eXt73** [comment]
> @eXt73 Please refrain from making nonsensical comments.

ok, but you can see that Nvidia's behavior [no answer to any questions asked in this topic] is very annoying here

**Comment by gabriele2000** [comment]
> ok, but you can see that Nvidia's behavior [no answer to any questions asked in this topic] is very annoying here

Heh, first time?
One day everyone will realize.

**Comment by eXt73** [comment]
The Cyberpunk 2077: Phantom Liberty expansion has been released, and there is support for NVIDIA DLSS 3.5 Ray Reconstruction ... and of course it does not work on Linux - this is just a supplement to the above-mentioned report ... Nvidia shows that it has us ...

**Comment by gabriele2000** [comment]
> The Cyberpunk 2077: Phantom Liberty expansion has been released, and there is support for NVIDIA DLSS 3.5 Ray Reconstruction ... and of course it does not work on Linux - this is just a 'supplement' to the above-mentioned report ... Nvidia shows that it has us ...

I'm baffled about the need of showing "proof" that nvidia doesn't care at all about Linux support, since it's SO EVIDENT

**Comment by K0bin** [comment]
@eXt73 As far as I know, ray reconstruction works. You just need to make NVAPI report a higher driver version and enable path tracing.

**Comment by Saancreed** [comment]
:point_up: This is correct, @eXt73 you need make sure you don't have outdated driver that doesn't support Ray Reconstruction (find `_nvngx.dll` file that comes with it and run `strings _nvngx.dll | grep dlss` on it, if it mentions `dlssd` then you are good to go) and export `DXVK_NVAPI_DRIVER_VERSION=53742` (until Nvidia releases 545 driver for Linux).

The requirement to enable Path Tracing to use Ray Reconstruction is the same as on Windows.

**Comment by eXt73** [comment]
> ☝️ This is correct, @eXt73 you need make sure you don't have outdated driver that doesn't support Ray Reconstruction (find `_nvngx.dll` file that comes with it and run `strings _nvngx.dll | grep dlss` on it, if it mentions `dlssd` then you are good to go) and export `DXVK_NVAPI_DRIVER_VERSION=53742` (until Nvidia releases 545 driver for Linux).
> 
> The requirement to enable Path Tracing to use Ray Reconstruction is the same as on Windows.
> 
> In fact, support for Ray Reconstruction in dxvk-nvapi has been implemented in [b4b33c2](https://github.com/jp7677/dxvk-nvapi/commit/b4b33c2abd2ba8aa70fa8081459bd2e511ac61ff) and merged over a month ago.

You're right ;) I'm honoring Nvidia here... the entry: DXVK_NVAPI_DRIVER_VERSION=53742 works as it should. Thanks.

Best regards

**Comment by eXt73** [comment]
Maybe > since Nvidia is still not kind enough to provide us with FG... maybe it would be possible to use the code of this modification? ;)

[Nvidia DLSS-G Frame Generation](https://www.nexusmods.com/site/mods/738?tab=files)

[dlssg-to-fsr3](https://github.com/Nukem9/dlssg-to-fsr3)

**Comment by jp7677** [comment]
> Maybe > since Nvidia is still not kind enough to provide us with FG... maybe it would be possible to use the code of this modification? ;)

See https://github.com/jp7677/dxvk-nvapi/issues/106

**Comment by Saancreed** [comment]
Well, yes? If NVIDIA didn't care then why would they bother [adding necessary work on vkd3d-proton side](https://github.com/HansKristian-Work/vkd3d-proton/pull/2053) to support this in the future?

Let's stay calm for a little longer, this stuff takes (a lot of) time.

**Comment by gilvbp** [comment]
Hi guys, you could test this version: https://www.mediafire.com/file/lv7d8jci0gyf6z0/proton_dlssfg.tar.zst/file

**Comment by jp7677** [comment]
@gilvbp Please don't distribute binaries here from non GitHub CI locations.

**Comment by shelterx** [comment]
I know it's under review but I tried the vkd3d-proton/dxvk-nvapi commits.
Don't have many FG-games but i tired Ghost of Tsushima, seems like it needs DXVK_NVAPI_GPU_ARCH=AD100  but enabling DLSS-FG makes the screen go black.

**Comment by Blisto91** [comment]
You need the Proton Wine commit too

**Comment by Saancreed** [comment]
> Ghost of Tsushima

@shelterx That's the one game for which [we _hide_ Ada GPUs by default](https://github.com/jp7677/dxvk-nvapi/pull/186) because of a bug in some external middleware it uses, so requiring the override there is currently expected. I haven't asked around for some time now whether it had been fixed or not but GoT should probably be given another look once DLFG support is merged.

**Comment by shelterx** [comment]
> > Ghost of Tsushima
> 
> @shelterx That's the one game for which [we _hide_ Ada GPUs by default](https://github.com/jp7677/dxvk-nvapi/pull/186) because of a bug in some external middleware it uses, so requiring the override there is currently expected. I haven't asked around for some time now whether it had been fixed or not but GoT should probably be given another look once DLFG support is merged.

Yeah, I remember that. that's why I tried AD100 instead.

I'm not sure if it's my proton but FYI, Black Myth says it need HAGS to be enabled. I used to have the HAGS patch but maybe it got removed...

What proton patch is needed then?

**Comment by Blisto91** [comment]
https://github.com/ValveSoftware/wine/pull/259

dlls also need to be placed in the prefix system32/syswow64

**Comment by shelterx** [comment]
Well, I think I got a working proton now so tried again GoT, FG works but it does not run well with AD100 exposed, but we knew that.
Also tried Black Myth benchmark, still complains about HAGS.
Was going to try Forza Horzion 5 too but that does not run under Wayland on 560.
